### PR TITLE
EREGCSC-2317 Update index endpoint

### PR DESCRIPTION
**Pull request by thwalker6**
Resolves # EREGCSC-2317

**Description-**

Adds update content update endpoint.  Currently there s no endpoint to update the index from the text extractor.

**This pull request changes...**

- Adds endpoint to update contentindex

**Steps to manually verify this change...**

1. There is currently a unit test to check.  currently the endpoint is more or less useless until the next story is added of updating the text extractor to update the content.

### CODE REVIEW

File: solution/backend/content_search/views.py [hunk 1]
```diff
@@ -95,3 +98,22 @@ def list(self, request):
...
         else:
             return Response(serializer.data)
+
+
+class PostContentTextViewset(APIView):
+    authentication_classes = [SettingsAuthentication]
+    permission_classes = [IsAuthenticatedOrReadOnly]
+
+    @extend_schema(
+        description="Retrieve list sof uploaded files",
```
Comment (peggles2) [inline solution/backend/content_search/views.py:108]: check the spelling here.

File: .github/workflows/deploy-experimental.yml [hunk 2]
```diff
@@ -174,6 +174,8 @@ jobs:
       DB_USER: eregs
       DB_PASSWORD: sgere
       DB_PORT: 5432
+      HTTP_AUTH_USER: eregs
+      HTTP_AUTH_PASSWORD: password
     services:
```
Comment (peggles2) [inline .github/workflows/deploy-experimental.yml:179]: what is this used for?

**Review by peggles2 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-experimental.yml (hunk 2 above)
Comment (thwalker6) [reply .github/workflows/deploy-experimental.yml:179]: Dummy environment variables for the unit test to check authentication on the endpoint.  If they do not have proper credentials they cannot access the endpoint.

File: solution/backend/content_search/tests/test_post.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                    file.locations.set([self.location2])
+                    file.subjects.set([self.subject2])
+                    file.category = category
+                    file.save()
+
+        index_group(SupplementalContent.objects.all())
+
+    def test_update_content(self):
+        content = ContentIndex.objects.first()
+        print(content.uid)
```
[the remark this answers is not in the document]
Comment (thwalker6) [reply solution/backend/content_search/tests/test_post.py]: removed

File: solution/backend/content_search/views.py (hunk 1 above)
Comment (thwalker6) [reply solution/backend/content_search/views.py]: changed

**Review by peggles2 — APPROVED** [review]
LGTM

**Pull request merged**